web/app/components/i18n.tsx

```diff
@@ -11,16 +11,20 @@ export type II18nProps = {
   locale: Locale
   dictionary: Record<string, any>
   children: React.ReactNode
-  setLocaleOnClient: (locale: Locale) => void
 }
 const I18n: FC<II18nProps> = ({
   dictionary,
   children,
+  locale,
 }) => {
-  const locale = getLocaleOnClient()
+  const clientLocale = getLocaleOnClient()
+
+  // force to change language if server have the locale to resolve the UI hydrate error
+  locale && changeLanguage(locale)
+
   return (
     <I18NContext.Provider value={{
-      locale,
+      locale: locale || clientLocale,
       i18n: dictionary,
       setLocaleOnClient,
     }}>
```

Comment:
**Suggestion:** Avoid calling `changeLanguage` directly in the component render. It's better to use `useEffect` hook to handle side effects like changing the language. This will ensure that the language is changed only when the `locale` prop changes.
```suggestion
const I18n: FC<II18nProps> = ({
  dictionary,
  children,
  locale,
}) => {
  const clientLocale = getLocaleOnClient()

  useEffect(() => {
    // force to change language if server have the locale to resolve the UI hydrate error
    locale && changeLanguage(locale)
  }, [locale])

  return (
    <I18NContext.Provider value={{
      locale: locale || clientLocale,
      i18n: dictionary,
      setLocaleOnClient,
    }}>
```